src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -0,0 +1,165 @@
+
+        // Retrieves the invocation expression node and the invocation argument list
+        protected abstract bool TryGetExpressionAndArguments(
+            SyntaxNode invocationNode,
+            [NotNullWhen(returnValue: true)] out SyntaxNode? expression,
+            [NotNullWhen(returnValue: true)] out List<SyntaxNode>? arguments);
+
+        // Verifies if the specified argument was passed with an explicit name.
+        protected abstract bool IsArgumentNamed(IArgumentOperation argumentOperation);
+
+        // Retrieves the invocation expression for a conditional operation, which consists of the dot and the method name.
+        protected abstract SyntaxNode GetConditionalOperationInvocationExpression(SyntaxNode invocationNode);
+
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(ForwardCancellationTokenToInvocationsAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (!(root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node))
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            // The analyzer created the diagnostic on the IdentifierNameSyntax, and the parent is the actual invocation
+            if (!TryGetInvocation(model, node, ct, out IInvocationOperation? invocation))
+            {
+                return;
+            }
+
+            ImmutableDictionary<string, string>? properties = context.Diagnostics[0].Properties;
+
+            // The name that identifies the object that is to be passed
+            if (!properties.TryGetValue(ForwardCancellationTokenToInvocationsAnalyzer.ArgumentName, out string argumentName) || string.IsNullOrEmpty(argumentName))
+            {
+                return;
+            }
+
+            // If the invocation requires the token to be passed with a name, use this
+            if (!properties.TryGetValue(ForwardCancellationTokenToInvocationsAnalyzer.ParameterName, out string parameterName))
+            {
+                return;
+            }
+
+            string title = MicrosoftNetCoreAnalyzersResources.ForwardCancellationTokenToInvocationsTitle;
+
+            if (!TryGenerateNewDocumentRoot(doc, root, invocation, argumentName, parameterName, out SyntaxNode? newRoot))
+            {
+                return;
+            }
+
+            Task<Document> createChangedDocument(CancellationToken _) =>
+                Task.FromResult(doc.WithSyntaxRoot(newRoot));
```

Comment:
We should avoid creating any new syntax nodes in RegisterCodeFixesAsync. This work should be delayed to create changed document callback in the code action when user actually applies the code fix:
```suggestion
            if (!TryGetExpressionAndArguments(invocation.Syntax, out _, out _)))
            {
                return;
            }
            
            Task<Document> createChangedDocument(CancellationToken _) =>
                Task.FromResult(doc.WithSyntaxRoot(TryGenerateNewDocumentRoot(doc, root, invocation, argumentName, parameterName, out SyntaxNode? newRoot)));
```

Reply:
Gotcha. Let me adjust the code accordingly.

Reply:
Fixed.